src/plugins/chart_expressions/expression_xy/public/components/annotations.tsx

```diff
@@ -229,6 +259,39 @@ export const Annotations = ({
           />
         );
       })}
+      {rangeAnnotations.map(({ label, time, color, endTime, outside }) => {
+        const id = snakeCase(label);
+        return (
+          <RectAnnotation
+            id={id}
+            key={id}
+            customTooltipDetails={() => (
+              <>
+                <EuiText size="xs">
+                  <h4>
+                    {formatter
+                      ? `${formatter.convert(time)} — ${formatter?.convert(endTime)}`
+                      : `${moment(time).toISOString()} — ${moment(endTime).toISOString()}`}
+                  </h4>
+                </EuiText>
+                <EuiText size="xs">{label}</EuiText>
+              </>
+            )}
```

Comment:
That's because tooltip for rectAnnotations doesn't have a `header` so I was trying to solve it in a different way. Pinging @markov00 again to see if I am not missing something.

Reply:
customTooltip for annotation are freeform actually, but you can just use this CSS for that:
```
 .echTooltip {
  @include euiToolTipStyle;
  @include euiFontSizeXS;
  padding: 0;
  transition: opacity $euiAnimSpeedNormal;
  pointer-events: none;
  user-select: none;
  max-width: 256px;

  &__header {
    @include euiToolTipTitle;
    margin-bottom: 0;
    padding: $euiSizeXS ($euiSizeXS * 2);
  }
```